modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java

```diff
@@ -3113,6 +3114,37 @@ public CodegenModel fromModel(String name, Schema schema) {
         return m;
     }
 
+    protected static void setEnumDiscriminatorDefaultValue(CodegenModel model) {
+        if (model.discriminator == null) {
+            return;
+        }
+        String discPropName = model.discriminator.getPropertyBaseName();
+        Stream.of(model.requiredVars, model.vars, model.allVars, model.readWriteVars)
+                .flatMap(List::stream)
+                .filter(v -> discPropName.equals(v.baseName))
+                .forEach(v -> v.defaultValue = getEnumValueForProperty(model.schemaName, model.discriminator, v));
+    }
+
+    protected static String getEnumValueForProperty(
+            String modelName, CodegenDiscriminator discriminator, CodegenProperty var) {
+        if (!discriminator.getIsEnum() && !var.isEnum) {
+            return null;
```

Comment:
Thanks for pointing this out. I changed it to use the existing defaultValue instead of null.

If I set defaultValue to the enum discriminator in `DefaultCodegen` then there is only this change to `dart-dio` when I generate the samples:
```diff
diff --git a/samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/doc/ChildWithNullable.md b/samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/
doc/ChildWithNullable.md
index 770494fcf4c..d3e5b182076 100644
--- a/samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/doc/ChildWithNullable.md
+++ b/samples/openapi3/client/petstore/dart-dio/petstore_client_lib_fake/doc/ChildWithNullable.md
@@ -8,7 +8,7 @@ import 'package:openapi/api.dart';
 ## Properties
 Name | Type | Description | Notes
 ------------ | ------------- | ------------- | -------------
-**type** | **String** |  | [optional]
+**type** | **String** |  | [optional] [default to ChildWithNullableTypeEnum.childWithNullable]
 **nullableProperty** | **String** |  | [optional]
 **otherProperty** | **String** |  | [optional]
 ```
 
 I am not sure if this change is fine or if I should just keep it to Java and C# for now, but this could also be a problem that needs to be solved for other languages.